### ISSUE TITLE
support wayland windowing system on linux

### ISSUE THREAD

**Issue by neurlang** [danfragoso/thdwb#46]
I've forked the browser and added support for the modern wayland linux-based windowing system.
It now ships as the [go-wayland-browser demo](https://github.com/neurlang/wayland/tree/master/go-wayland-browser) of my wayland repository.

What works:
* buttons, icons
* keyboard input including unicode
* scrolling

What doesn't work yet:
* popup menus, i intend to add these
* input methods (virtual keyboards for CJK and emojis etc)
* overlays - i don't know what it is

Differences:
* there is no glfw anymore, so untested on windows os (however it might work using the windows compatibility in my lib)
* no open gl dependency anymore, drawing via pure shm on cpu only (shared memory)
* no shaders lol
* compiles without cgo, so can use additional static analysis tools like go build race detector

The changes are confined to the mustard package. The changes to the browser package are minimal.

**Comment by neurlang** [comment]
![Screenshot from 2024-02-10 15-33-24](https://github.com/danfragoso/thdwb/assets/77860779/5852d8ec-c63c-48ba-b3d4-37c143c55297)

**Comment by neurlang** [comment]
fixed colors, fixed popup menu (overlays) to appear (in the wrong corner for now)

what doesn't work:
- key repeat

![Screenshot from 2024-02-11 14-41-29](https://github.com/danfragoso/thdwb/assets/77860779/63cc0574-d778-4dbe-a3ea-d243ca2780e8)

**Comment by neurlang** [comment]
what works:
- can access debug mode and tree mode

![Screenshot from 2024-02-11 20-35-27](https://github.com/danfragoso/thdwb/assets/77860779/7070b5c0-f297-4a51-ba9a-2a9114554449)

**Comment by danfragoso** [comment]
Thank you for doing this @neurlang! I really appreciate you taking the time to do it, I'll update the repo to mention that this fork is available for Wayland, if you want to, we can work together on merging it to repo

**Comment by neurlang** [comment]
Yeah, definitely it can be merged (when mature). My fork of the mustard package can be added to your mustard package to be activated the build tag wayland. We did this in the past in the vulkan repo and it worked.

This means that anyone will be able to compile the browser with the wayland mustard using:
```
go build -tags wayland
```
Whereas the old glfw mustard will still work like this:
```
go build
```

**Comment by neurlang** [comment]
I've modernized it, ported it to new gg (`github.com/gogpu/gg`) etc. as a bonus there is now body bgcolor support.

https://github.com/user-attachments/assets/a884cecd-98ee-4a1f-809c-8b1d999e54da

Should work on linux, windows, probably mac.
